task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,57 @@
+import re
+import secrets
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+BASE_58 = '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
+UUID_BYTES = 6
+MAX_POD_NAME_LENGTH = 253
+REGEX_MATCH = '[a-z0-9]([.-a-z0-9]*[a-z0-9])?'
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    def __invariant__(conf):
+        return(
+            (
+                len(conf.pod_name) > MAX_POD_NAME_LENGTH,
+                (
+                    f'Invalid format for pod_name {conf.pod_name}. ',
+                    f'Pod name must have up to {MAX_POD_NAME_LENGTH}.'
+                )
+            ),
+            (
+                re.match(REGEX_MATCH, conf.pod_name),
+                (
+                    f'Invalid format for pod_name {conf.pod_name}. ',
+                    f'Must comply with Kubernetes pod naming standards.'
+                )
+            )
+        )
+    node_selector = field(type=PMap)
+    containers = field(type=PVector)
+    # Hardcoded for the time being
+    restart_policy = "Never"
+    # By default, the retrying executor retries 3 times. This task option
+    # overrides the executor setting.
+    retries = field(type=int,
+                    factory=int,
+                    mandatory=False,
+                    invariant=lambda r: (r >= 0, 'retries >= 0'))
+    volumes = field(type=PVector)
+
+    @property
+    def pod_name(self) -> str:
+        return f'{self.name}.{self.uuid}'
+
+    def set_pod_name(self, pod_name: str):
+        try:
+            name, = pod_name.rsplit('.', maxsplit=1)
+            uuid = ''.join(secrets.choice(BASE_58) for i in range(UUID_BYTES))
```

Comment:
we don't want to generate a new unique suffix here since callers of this function are already going to provide one for us so we want something like 
```suggestion
            name, uuid = pod_name.rsplit('.', maxsplit=1)
```

Reply:
after that, the code that generates the "uuid" (i.e.,`''.join(secrets.choice(BASE_58) for i in range(UUID_BYTES))`) should probably be a function that we then use in the uuid field

e.g.:
```
def _generate_pod_suffix() -> str:
    return ''.join(secrets.choice(POD_SUFFIX_ALPHABET) for i in range(POD_SUFFIX_LENGTH))
    

class KubernetesTaskConfig(DefaultTaskConfigInterface):
    uuid = field(type=str, initial=_generate_pod_suffix)
    ...
```

since we only want task-proc to generate a "uuid" when creating a brand new task, not when editing an existing one